### PR TITLE
Added OpenACC directives to Convection and New Tiedtke model, for GPU execution.

### PR DESCRIPTION
**Pull request by cponder**
Added OpenACC directives from 6.x-openacc version to manage Host <-> GPU memory movement, to enable convection model to run on the GPU.

### CODE REVIEW

**Comment by cponder** [comment]
The second commit was to remove deprecated array names from the Device -> Host update.

File: src/core_atmosphere/physics/mpas_atmphys_driver_convection.F
Comment (supreethms1809) [inline src/core_atmosphere/physics/mpas_atmphys_driver_convection.F:116]: @cponder I think we need to get rid of all these debug messages by submitting one more commit

Comment (supreethms1809) [reply src/core_atmosphere/physics/mpas_atmphys_driver_convection.F:116]: There are multiple debug messages below. the above comment applies to all of them

File: src/core_atmosphere/physics/physics_wrf/module_cu_ntiedtke.F
Comment (supreethms1809) [inline src/core_atmosphere/physics/physics_wrf/module_cu_ntiedtke.F:342]: @cponder instead of commenting the deleted code in V7. I think we should delete it as well

**Review by supreethms1809 — CHANGES_REQUESTED** [review]
Apart from two general comments the openacc port looks good. Will wait for @Pranay-Reddy-Kommera confirmation to start the runs and validation.

**Comment by mgduda** [comment]
Following a discussion with @raghuraj19, I've added the "Don't merge" label: we need to wait until @Pranay-Reddy-Kommera's PRs have been merge before merging this PR.

**Comment by cponder** [comment]
I've had to re-work the changes to go with #909 & #910.
I wrote a new PR #927 and am closing this one.

**Pull request closed without merging**
